### PR TITLE
[CodeSynthesis] Improve synthesized `Decodable.init(from:)` for enums

### PR DESCRIPTION
**Pull request by rintaro**
* Don't use `unsafelyUnwrapped` which is usually not recommended
* Don't access `container.allKeys` multiple times as it's a computed property

Before:
```swift
  guard container.allKeys.count == 1 else { throw ... }
  switch container.allKeys.first.unsafelyUnwrapped { ... }
```

After:
```swift
  var allKeys = ArraySlice(container.allKeys)
  guard let onlyKey = allKeys.popFirst(), allKeys.isEmpty else { throw ... }
  switch onlyKey { ... }
```

rdar://89150202

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by rintaro** [comment]
@swift-ci Please test

File: test/refactoring/AddCodableImplementation/Outputs/enum/codable.swift.expected [hunk 1]
```diff
@@ -18,10 +18,12 @@ enum Payload: Codable {
...
     let container = try decoder.container(keyedBy: Payload.CodingKeys.self)
 
-    guard container.allKeys.count == 1 else {
+    var allKeys = ArraySlice(container.allKeys)
+
+    guard let onlyKey = allKeys.popFirst(), allKeys.isEmpty else {
       throw DecodingError.typeMismatch(Payload.self, DecodingError.Context.init(codingPath: container.codingPath, debugDescription: "Invalid number of keys found, expected one.", underlyingError: nil))
     }
-    switch container.allKeys.first.unsafelyUnwrapped {
+    switch onlyKey {
```
Comment (louisdh) [inline test/refactoring/AddCodableImplementation/Outputs/enum/codable.swift.expected:26]: Perfect!

File: test/refactoring/AddCodableImplementation/Outputs/enum/codable.swift.expected [hunk 2]
```diff
@@ -18,10 +18,12 @@ enum Payload: Codable {
   init(from decoder: Decoder) throws {
     let container = try decoder.container(keyedBy: Payload.CodingKeys.self)
 
-    guard container.allKeys.count == 1 else {
+    var allKeys = ArraySlice(container.allKeys)
+
+    guard let onlyKey = allKeys.popFirst(), allKeys.isEmpty else {
```
Comment (louisdh) [inline test/refactoring/AddCodableImplementation/Outputs/enum/codable.swift.expected:23]: Perfect!

I was confused at first seeing `allKeys.isEmpty`. Then I realized it's mutating `allKeys` with the `popFirst()` (I initially just read it as _accessing_ the first key, not removing it from the array). 

Seems like a much more efficient (and safe) way for checking that's there's only 1 key.

**Review by louisdh — APPROVED** [review]
Thank you!

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by airspeedswift — APPROVED** [review]
looks good, thanks!

**Pull request merged**